components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/main/java/org/wso2/carbon/apimgt/rest/api/core/utils/PolicyExportManager.java

```diff
@@ -0,0 +1,242 @@
+            APIFileUtils.createDirectory(exportedPoliciesDirName);
+            if (!apiPolicies.isEmpty()) {
+                for (Map<String, String> map : getApiPolicySiddhiApps(apiPolicies)) {
+                    prepareFile(exportedPoliciesDirName, map);
+                }
+            }
+            if (!applicationPolicies.isEmpty()) {
+                prepareFile(exportedPoliciesDirName, getAppPolicySiddhiApps(applicationPolicies));
+            }
+            if (!subscriptionPolicies.isEmpty()) {
+                prepareFile(exportedPoliciesDirName, getSubscriptionPolicySiddhiApps(subscriptionPolicies));
+            }
+            if (!customPolicies.isEmpty()) {
+                prepareFile(exportedPoliciesDirName, getCustomPolicySiddhiApps(customPolicies));
+            }
+            //create archive and get the archive location
+            String zippedFilePath = createArchiveFromPolicies(rootFolder, archiveDir, archiveName);
+            APIFileUtils.deleteDirectory(rootFolder);
+            return zippedFilePath;
+        } catch (APIManagementException e) {
+            String errorMessage = "Error while exporting policies";
+            log.error(errorMessage, e);
+            throw new APIManagementException(errorMessage, e);
+        }
+    }
+
+    /**
+     * Create Archive from policies.
+     *
+     * @param sourceDir   Source directory containing execution plans
+     * @param archiveDir  Archive containing directory
+     * @param archiveName name of the archive
+     * @return String object of the file path
+     * @throws APIMgtDAOException if creating archive failed
+     */
+    private String createArchiveFromPolicies(String sourceDir, String archiveDir, String archiveName)
+            throws APIMgtDAOException {
+        if (log.isDebugEnabled()) {
+            log.debug("Creating archive for execution plans.");
+        }
+        APIFileUtils.archiveDirectory(sourceDir, archiveDir, archiveName);
+        String archivePath = archiveDir + File.separator + archiveName + ZIP;
+        log.info("Created archive for execution plans: " + archivePath);
+        return archivePath;
+    }
+
+    /**
+     * write execution plan to a file.
+     *
+     * @param exportDir location contains the files
+     * @param policies  Map of execution plan names and execution plans
+     */
+    private void prepareFile(String exportDir, Map<String, String> policies) {
+        if (log.isDebugEnabled()) {
+            log.debug("Writing files for execution plans.");
+        }
+        policies.forEach((name, siddhiApp) -> {
+            try {
+                APIFileUtils.createFile(exportDir + File.separator + name);
+                APIFileUtils.writeToFile(exportDir + File.separator + name, siddhiApp);
+            } catch (APIMgtDAOException e) {
+                String errorMsg = "Error while preparing the file with policy name: " + name;
+                log.error(errorMsg, e);
```

Comment:
we should throw here if an error occurs. if not there might be some policies missing.